packages/openapi-v3-types/src/openapi-v3-spec-types.ts

```diff
@@ -0,0 +1,395 @@
+ * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#parameter-object
+ */
+export interface ParameterObject extends ISpecificationExtension {
+  name: string;
+  in: ParameterLocation;
+  description?: string;
+  required?: boolean;
+  deprecated?: boolean;
+  allowEmptyValue?: boolean;
+
+  style?: ParameterStyle;
+  explode?: boolean;
+  allowReserved?: boolean;
+  schema?: SchemaObject | ReferenceObject;
+  examples?: {[param: string]: ExampleObject | ReferenceObject};
+  example?: ExtensionValue;
+  content?: ContentObject;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface ExampleObject {
+  summary?: string;
+  description?: string;
+  value?: ExtensionValue;
+  externalValue?: string;
+  [property: string]: ExtensionValue; // Hack for allowing ISpecificationExtension
+}
+
+export interface ReferenceObject {
+  $ref: string;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `SchemaObject`
+ */
+export interface DiscriminatorObject {
+  propertyName: string;
+  mapping?: {[key: string]: string};
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `SchemaObject`
+ */
+export interface XmlObject extends ISpecificationExtension {
+  name?: string;
+  namespace?: string;
+  prefix?: string;
+  attribute?: boolean;
+  wrapped?: boolean;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `SchemaObject`
+ */
+export interface ExternalDocumentationObject extends ISpecificationExtension {
+  description?: string;
+  url: string;
+}
+
+//  Specification Extensions
+//   ^x-
+export interface ISpecificationExtension {
+  // Cannot constraint to "^x-" but can filter them later to access to them
+  [extensionName: string]: ExtensionValue;
+}
+
+/**
+ * Maps names to a given type of values
+ */
+export interface MapObject<T> {
+  /**
+   * Maps between a name and object
+   */
+  [name: string]: T;
+}
+
+/**
+ * Lists the available scopes for an OAuth2 security scheme.
+ */
+export interface ScopesObject
+  extends MapObject<string>,
+    ISpecificationExtension {
+  /**
+   * Maps between a name of a scope to a short description of it (as the value
+   * of the property).
+   */
+  [name: string]: string;
+}
+
+/**
+ * Copied from swagger 2 spec, don't find our code use it.
+ * A declaration of the security schemes available to be used in the
+ * specification. This does not enforce the security schemes on the operations
+ * and only serves to provide the relevant details for each scheme.
+ */
+export interface SecurityDefinitionsObject
+  extends MapObject<OAS3.SecuritySchemeObject> {
+  /**
+   * A single security scheme definition, mapping a "name" to the scheme it
+   * defines.
+   */
+  [name: string]: OAS3.SecuritySchemeObject;
+}
+
+/**
+ * Copied from swagger 2 spec, don't find our code uses it.
```

Comment:
Please rephrase this. We're _extending_ the Swagger specification to allow additional parameters.

Reply:
ah...sorry I should not add it as part of the jsdoc.
For those commented code, I am actually expecting opinions for them. I will check though the whole to make them clearer.